### ISSUE TITLE
Color code async members differently in the analyzer syntax highlighting output

### ISSUE THREAD

**Issue by jacob314** [dart-lang/sdk#46211]
In a discussion of https://dart-lang.github.io/linter/lints/unawaited_futures.html, it came up that it isn't obvious from the code editor if functions being called are async unless there is an `await` or `unawaited` near the method call.
Perhaps we  can color code async methods being called differently so that you can immediately see the "color of each function".

Blog post from @munificent from 2015 for inspiration. https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/

**Comment by bwilkerson** [comment]
I suspect we would want to color code functions that return a `Future` or `FutureOr`, not those marked as `async`. Is that correct?

I'm not necessarily opposed to the idea, but I'm not sure to what degree doing so would help users given that we already have a lint for it.

Also, server's legacy protocol passes back an enum describing the semantics of a range of characters, but the real job of translating that into a color is done in the client. We'd need to updated the clients (primarily IntelliJ) as part of this task. We're in a little better place with LSP because they've already built this into the protocol, so clients might already be using it. We'd just need to add `SemanticTokenModifiers.async` to the `function` and `method` types when appropriate.

**Comment by jacob314** [comment]
Correct. You would want to color code functions based on return value.
I agree this is P3 as we already have the lint. The lint is great but can't capture all the edge cases. Additionally, even with the lint it is useful to be able to visually see where there is async code.
For example
```dart
Future<void> someAsyncMethod() async {
  final foo = someOtherAsyncMethod1();
  someMethodTakingObject(someOtherAsyncMethod2(foo));
}
```
Here the lint can't fire to indicate whether `someOtherAsyncMethod1` or `someOtherAsyncMethod2` were unawaited.

**Comment by bwilkerson** [comment]
> Here the lint can't fire to indicate whether ...

Actually, the rule _could_ produce a lint, but we chose not to produce one in that case because there is another issue (non-assignability), and it is the other issue that needs to be addressed first (and once it's addressed there likely won't be an un-awaited future).
